ibtk/include/ibtk/CCPoissonHypreLevelSolver.h

```diff
@@ -300,40 +301,40 @@ class CCPoissonHypreLevelSolver : public LinearSolver, public PoissonSolver
     /*!
      * \brief Associated hierarchy.
      */
-    SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM> > d_hierarchy;
+    SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM> > d_hierarchy = nullptr;
 
     /*!
      * \brief Associated patch level and C-F boundary (for level numbers > 0).
      */
-    int d_level_num;
-    SAMRAI::tbox::Pointer<SAMRAI::hier::PatchLevel<NDIM> > d_level;
-    SAMRAI::tbox::Pointer<SAMRAI::hier::CoarseFineBoundary<NDIM> > d_cf_boundary;
+    int d_level_num = -1;
+    SAMRAI::tbox::Pointer<SAMRAI::hier::PatchLevel<NDIM> > d_level = nullptr;
+    SAMRAI::tbox::Pointer<SAMRAI::hier::CoarseFineBoundary<NDIM> > d_cf_boundary = nullptr;
 
     /*!
      * \name Problem specification.
      */
-    bool d_grid_aligned_anisotropy;
+    bool d_grid_aligned_anisotropy = true;
 
     /*!
      * \name hypre objects.
      */
     //\{
-    unsigned int d_depth;
-    HYPRE_StructGrid d_grid;
-    HYPRE_StructStencil d_stencil;
-    std::vector<HYPRE_StructMatrix> d_matrices;
-    std::vector<HYPRE_StructVector> d_rhs_vecs, d_sol_vecs;
-    std::vector<HYPRE_StructSolver> d_solvers, d_preconds;
-    std::vector<SAMRAI::hier::Index<NDIM> > d_stencil_offsets;
+    unsigned int d_depth = 0;
+    HYPRE_StructGrid d_grid = nullptr;
+    HYPRE_StructStencil d_stencil = nullptr;
+    std::vector<HYPRE_StructMatrix> d_matrices = {};
+    std::vector<HYPRE_StructVector> d_rhs_vecs = {}, d_sol_vecs = {};
+    std::vector<HYPRE_StructSolver> d_solvers = {}, d_preconds = {};
+    std::vector<SAMRAI::hier::Index<NDIM> > d_stencil_offsets = {};
 
-    std::string d_solver_type, d_precond_type;
-    int d_rel_change;
-    int d_num_pre_relax_steps, d_num_post_relax_steps;
-    int d_memory_use;
+    std::string d_solver_type = "PFMG", d_precond_type = "none";
+    int d_rel_change = 0;
+    int d_num_pre_relax_steps = 1, d_num_post_relax_steps = 1;
+    int d_memory_use = 0;
     int d_rap_type;
```

Comment:
Do these guys not get default values?

Reply:
I only added default values to members who got default values on initialization. For example, here d_rap_type was initialized with an input to the constructor. Others might not have been initialized at all.

I can go back through and initialize on declaration all member variables default values.

Reply:
I think that it is fine to omit things that do not always get default values.